packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/TreeControllerIntegrationTests.java

```diff
@@ -0,0 +1,248 @@
+    );
+
+    private final TreeItemMatcher documentHasNoObject = new TreeItemMatcher(
+            tree -> tree.getChildren().get(0),
+            treeItem -> !treeItem.isHasChildren()
+    );
+
+    @Autowired
+    private IGraphQLRequestor graphQLRequestor;
+
+    @Autowired
+    private IEditingContextSearchService editingContextSearchService;
+
+    @Autowired
+    private IIdentityService identityService;
+
+    @Test
+    @DisplayName("Given a project, when we subscribe to the tree events of its explorer, then the tree of the explorer is sent")
+    @Sql(scripts = {"/scripts/initialize.sql"}, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = {"/scripts/cleanup.sql"}, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD, config = @SqlConfig(transactionMode = SqlConfig.TransactionMode.ISOLATED))
+    public void givenProjectWhenWeSubscribeToTreeEventsOfItsExplorerThenTheTreeOfTheExplorerIsSent() {
+        var input = new TreeEventInput(UUID.randomUUID(), TestIdentifiers.ECORE_SAMPLE_PROJECT.toString(), "explorer://", List.of());
+        var flux = this.graphQLRequestor.subscribe(GET_TREE_EVENT_SUBSCRIPTION, input);
+
+        Predicate<Object> projectContentMatcher = object -> Optional.of(object)
+                .filter(DataFetcherResult.class::isInstance)
+                .map(DataFetcherResult.class::cast)
+                .map(DataFetcherResult::getData)
+                .filter(TreeRefreshedEventPayload.class::isInstance)
+                .isPresent();
+
+        StepVerifier.create(flux)
+                .expectNextMatches(projectContentMatcher)
+                .thenCancel()
+                .verify(Duration.ofSeconds(10));
+    }
+
+    @Test
+    @DisplayName("Given a project, when we subscribe to the tree events of its explorer with everything expanded, then all objects are returned")
+    @Sql(scripts = {"/scripts/initialize.sql"}, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = {"/scripts/cleanup.sql"}, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD, config = @SqlConfig(transactionMode = SqlConfig.TransactionMode.ISOLATED))
+    public void givenProjectWhenWeSubscribeToTreeEventsOfItsExplorerWithEverythingExpandedThenAllObjectsAreReturned() {
+        var input = new TreeEventInput(UUID.randomUUID(), TestIdentifiers.STUDIO_PROJECT.toString(), "explorer://", List.of());
+        var flux = this.graphQLRequestor.subscribe(GET_TREE_EVENT_SUBSCRIPTION, input);
+
+        Predicate<Object> hasProjectContent = object -> Optional.of(object)
+                .filter(DataFetcherResult.class::isInstance)
+                .map(DataFetcherResult.class::cast)
+                .map(DataFetcherResult::getData)
+                .filter(TreeRefreshedEventPayload.class::isInstance)
+                .isPresent();
+
+        StepVerifier.create(flux)
+                .expectNextMatches(hasProjectContent)
+                .thenCancel()
+                .verify(Duration.ofSeconds(10));
```

Comment:
This does not seem to test what it says it tests. It does not provide anything in the `expanded` list, nor tests the actual content of the returned payload.

Reply:
Yes indeed, I restructured my tests and this one ended up being a copy of the previous one and the more complex behavior has been merged with the next tests since it was doing it anyway.